keps/sig-storage/0031-20181029-VolumeSubpathEnvExpansion-apichange.md

```diff
@@ -0,0 +1,351 @@
+      - command:
+        - /bin/sh
+        - -xc
+        - |
+          LOGDIR=/logs/${POD_NAMESPACE}/${POD_NAME}; mkdir -p ${LOGDIR} && ln -sfv ${LOGDIR} /workdir/logs && chmod -R ugo+wr ${LOGDIR}
+        env:
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.name
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+        image: alpine:3.5
+        name: prep-logs
+        volumeMounts:
+        - mountPath: /logs
+          name: logs
+        - mountPath: /workdir
+          name: workdir
+      volumes:
+      - emptyDir: {}
+        name: workdir
+      - hostPath:
+          path: /logs
+          type: ""
+        name: logs
+```
+
+## Workarounds - k8s >1.9.3
+
+This makes use of relative symbolic linking to the underlying operating system
+
+```
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: private-image-test-1
+  labels:
+    application: podtest
+spec:
+  replicas: 2
+  template:
+    metadata:
+      labels:
+        application: podtest
+      name: private-image-test-1
+    spec:
+      initContainers:
+      - command:
+        - /bin/sh
+        - -xc
+        - |
+          LOGDIR=/logs/${POD_NAMESPACE}/${POD_NAME}; mkdir -p ${LOGDIR} && mkdir /workdir/logs && ln -rsfv /var/lib/kubelet/pods/${POD_UID}/volumes/kubernetes.io~empty-dir/workdir/logs ${LOGDIR} && chmod -R ugo+wrx /pods
+        env:
+        - name: POD_UID
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.uid
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.name
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+        image: debian
+        imagePullPolicy: IfNotPresent
+        name: prep-logs
+        volumeMounts:
+        - mountPath: /logs
+          name: logs
+        - mountPath: /workdir
+          name: workdir
+        - mountPath: /pods
+          name: pods
+      containers:
+      - name: uses-private-image
+        image: busybox
+        env:
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.name
+        command: [ "sh", "-c", "while [ true ]; do echo 'Hello $(POD_NAME)'; sleep 10; done | tee -a /logs/hello.txt" ]
+        imagePullPolicy: Always
+        volumeMounts:
+        - mountPath: /logs
+          name: workdir
+          subPath: logs
+      volumes:
+      - name: workdir
+        emptyDir: {}
+      - name: logs
+        hostPath: 
+          path: /logs
+      - name: pods
+        hostPath: 
+          path: /var/lib/kubelet/pods
+```
+
+## Alternatives - using subPath directly
```

Comment:
@thockin I added a section here in the alternatives to describe the existing alpha feature
It was assumed that this albeit already implemented and commented on would still be considered as a viable alternative to the api change proposal. Happy to move it, if you feel that it would be more appropriate in the proposal section

Reply:
/cc @msau42

Reply:
got it.